trin-state/src/network.rs

```diff
@@ -72,20 +74,32 @@ impl StateNetwork {
         // Trigger bonding with bootnodes, at both the base layer and portal overlay.
         // The overlay ping via talkreq will trigger a session at the base layer, then
         // a session on the (overlay) portal network.
-        for enr in self.overlay.discovery.discv5.table_entries_enr() {
+        // Will return error (and end execution) if zero bootnodes succeed bonding.
+        let mut successfully_bonded_bootnode = false;
+        let bootnodes = self.overlay.discovery.discv5.table_entries_enr();
+        if bootnodes.is_empty() {
+            return Ok(());
```

Comment:
I'm not sure why this needs to be `return Ok(());`, whereas in the `if` clause on line 97/98 I can just do `Ok(())`. Compiler complains with `expected (), found enum Result`.